### PR TITLE
sub: Allow setting lavc txt_page special values via teletext_page

### PR DESCRIPTION
**Pull request by MoSal**
* Range of accepted values for teletext_page now include 0 and -1.
 * 0 means "subtitle" and -1 means "*".
 * Make 0 the default.

Read this before you submit this pull request:
https://github.com/mpv-player/mpv/blob/master/DOCS/contribute.md

Reading this link and following the rules will get your pull request reviewed
and merged faster. Nobody wants lazy pull requests.

### CODE REVIEW

**Comment by kasper93** [comment]
Isn't teletext page 100 considered the initial page? And that why it was default?

**Comment by llyyr** [comment]
> Isn't teletext page 100 considered the initial page? And that why it was default?

According to https://ffmpeg.org/ffmpeg-codecs.html#Options-9 the default value is `*` or -1, we default to 0 or `subtitle`.

**Comment by MoSal** [comment]
I chose 0 because I'm assuming subtitles is what most people care about. And actual teletext enjoyers have to already be familiar with `teletext-page`, so they can presumably figure things out on there own.

I'm fine with -1 as a default too.

File: DOCS/man/options.rst [hunk 1]
```diff
@@ -2755,9 +2755,12 @@ Subtitles
     of subtitles across seeks, so after a seek libass can't eliminate subtitle
     packets with the same ReadOrder as earlier packets.
 
-``--teletext-page=<1-999>``
+``--teletext-page=<-1-999>``
     This works for ``dvb_teletext`` subtitle streams, and if FFmpeg has been
```
Comment (sfan5) [inline DOCS/man/options.rst:2759]: while you're editing this could you add a real description? It doesn't actually say what the option does.

Comment (MoSal) [reply DOCS/man/options.rst:2764]: Something like the commit I just pushed?

**Review by sfan5 — APPROVED** [review]
LGTM

**Review by kasper93 — APPROVED** [review]
LGTM and works

**Comment by kasper93** [comment]
Although we could mention page `100` in documentation. `100` is "home" page of teletext. So it might be good to mention if someone want to read good old teletext, not only the subtitles.

**Comment by MoSal** [comment]
> Although we could mention page 100 in documentation. 100 is "home" page of teletext and default start page on any teletext aware hardware. So it might be good to mention if someone want to read good old teletext, not only the subtitles.

I actually already had this thought, but more in the context of mentioning the previous default.

Pushed a change that mentions this.

**Comment by sfan5** [comment]
The changed default also needs to be mentioned here: https://github.com/mpv-player/mpv/blob/master/DOCS/interface-changes.rst
(in the same commit that changes it)

**Comment by MoSal** [comment]
> The changed default also needs to be mentioned here: https://github.com/mpv-player/mpv/blob/master/DOCS/interface-changes.rst (in the same commit that changes it)

DONE

**Pull request merged**